### PR TITLE
Advanced imputation metrics

### PR DESCRIPTION
**Pull request by tcasstevens**
Added option to merge-gvcfs to provide a bedfile that subsets to multiple output VCF files

### CODE REVIEW

File: src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt [hunk 1]
```diff
@@ -124,6 +136,63 @@ private suspend fun writeOutputVCF(
...
+    var writer: VariantContextWriter? = null
+    for (deferred in variantContextChannel) {
+
+        val variantContexts = deferred.await()
+        variantContexts.forEach { variantContext ->
+
+            val currentPositionRange = PositionRange(variantContext.contig, variantContext.start, variantContext.end)
+
+            // If the current position range doesn't overlap with the current write position range
+            // then close the current writer and open a new one
```
Comment (lynnjo) [inline src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt:164]: What happens if the VariantContext record overlaps multiple ranges?  Should it get split, with parts written to range-specific files for all files it overlaps?

File: src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt [hunk 2]
```diff
@@ -16,7 +19,7 @@ private val myLogger = LogManager.getLogger("biokotlin.util.MergeGVCFUtils")
```
Comment (lynnjo) [inline src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt:19]: This file should have a header comment with a description of the scope of functionality that is included.

File: src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt [hunk 3]
```diff
@@ -124,6 +136,63 @@ private suspend fun writeOutputVCF(
...
+            if (writePositionRange == null || !writePositionRange!!.overlapping(currentPositionRange)) {
+
+                // Close the previous writer
+                writer?.close()
+                writer = null
+
+                // Find the range in the bedfile that overlaps with the current position range
+                // and create a new writer for that range. If no range is found, then don't create
+                // a writer.
+                writePositionRange = ranges.find { it.overlapping(currentPositionRange) }
```
Comment (lynnjo) [inline src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt:174]: writePositionRange could be multiple ranges?

**Review by smm477 — APPROVED** [review]

File: src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt (hunk 3 above)
Comment (tcasstevens) [reply src/main/kotlin/biokotlin/util/MergeGVCFUtils.kt:180]: These are SNPs, so only one position per variant

**Review by lynnjo — APPROVED** [review]

**Review by aberthel — APPROVED** [review]

**Review by btmonier — APPROVED** [review]

**Pull request merged**